### PR TITLE
Fixes #3890: Shared link bypasses biometrics

### LINKED ISSUE #3890: Home Screen Link or Shared link unlocks session
Edit: Using Version 7.0.13 (Build #323021954)
## Prerequisite
Press hamburger button > Settings > Privacy & Security > Switching Apps
Enable "Use fingerprint to unlock app"

### Steps to reproduce
- Start new Firefox Focus session
- visit any website (https://bing.com)
- press home key 
- Reopen focus 

`Unlock Firefox Focus` screen is displayed, allowing the user to unlock or create a new session

At this point, the session is 'locked'

Then, either
    Press home key, select previously created shortcut on the home screen
or
    Open another browser, and select share, and choose Focus as the destination

-------
At this point, the link is opened in Firefox Focus. 
***The trash/tabs button can be pressed to view other tabs from the otherwise locked session.***

I think rather Firefox Focus should open, with option for user to open the link in a new session, or to unlock the session and open the link.

### PR DESCRIPTION
**Pull request by sblatz**
### Demo

new behavior

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/biometrics/BiometricAuthenticationDialogFragment.kt [hunk 1]
```diff
@@ -80,11 +85,17 @@ class BiometricAuthenticationDialogFragment : AppCompatDialogFragment(), Lifecyc
     interface BiometricAuthenticationListener {
         fun onCreateNewSession()
```
Comment (ekager) [inline app/src/main/java/org/mozilla/focus/biometrics/BiometricAuthenticationDialogFragment.kt]: Can we rename this its not clear its involving biometric and not our sessionmanager sessions

File: app/src/main/java/org/mozilla/focus/biometrics/BiometricAuthenticationDialogFragment.kt [hunk 2]
```diff
@@ -80,11 +85,17 @@ class BiometricAuthenticationDialogFragment : AppCompatDialogFragment(), Lifecyc
...
         fun onAuthSuccess()
+        fun onCreateNewSessionWithLink()
```
Comment (ekager) [inline app/src/main/java/org/mozilla/focus/biometrics/BiometricAuthenticationDialogFragment.kt]: And rename this one :)

File: app/src/main/java/org/mozilla/focus/biometrics/BiometricAuthenticationDialogFragment.kt [hunk 3]
```diff
@@ -80,11 +85,17 @@ class BiometricAuthenticationDialogFragment : AppCompatDialogFragment(), Lifecyc
...
     }
 
-    fun newSessionButtonClicked() {
+    private fun newSessionButtonClicked() {
```
Comment (ekager) [inline app/src/main/java/org/mozilla/focus/biometrics/BiometricAuthenticationDialogFragment.kt]: rename something like "biometricNewSessionButtonClicked"

**Comment by colintheshots** [comment]
ktlint is failing:

/opt/focus-android/app/src/main/java/org/mozilla/focus/biometrics/BiometricAuthenticationHandler.kt:83:32: Parameter should be on a separate line (unless all parameters can fit a single line)
/opt/focus-android/app/src/main/java/org/mozilla/focus/biometrics/BiometricAuthenticationHandler.kt:84:32: Unexpected indentation (expected 8, actual 31)
/opt/focus-android/app/src/main/java/org/mozilla/focus/biometrics/BiometricAuthenticationHandler.kt:84:63: Missing newline before ")"

**Comment by sblatz** [comment]
Fixed nits and added [this line](https://github.com/mozilla-mobile/focus-android/pull/3910/files#diff-9f997079c3a3095aaff772f3c980dc79R299) to account for home screen icons. Ktlint is also passing (forgot to add the hook back when my laptop got wiped). Let me know if this is safe to merge!

**Pull request merged**